pkg/api/types.go

```diff
@@ -748,30 +748,56 @@ func (p *Properties) K8sOrchestratorName() string {
 	return ""
 }
 
-func (p *Properties) getAgentVMPrefix() string {
-	return p.K8sOrchestratorName() + "-agentpool-" + p.GetClusterID() + "-"
+// GetAgentPoolIndex returns the prefix of agentpool VMs given the pool index1
+func (p *Properties) GetAgentPoolIndex(a *AgentPoolProfile) int {
+	index := -1
+	for i, profile := range p.AgentPoolProfiles {
+		if profile.Name == a.Name {
+			index = i
+			break
+		}
+	}
+	return index
+}
+
+// GetAgentVMPrefix returns the prefix of agentpool VMs given the pool index
```

Comment:
Based on my previous comment, Let the godoc be " GetAgentVMPrefix returns the prefix of agentpool VMs given the agent pool profile name"